tests/models/perceiver/test_modeling_perceiver.py

```diff
@@ -727,11 +731,14 @@ def test_correct_missing_keys(self):
 
         for model_class in self.all_model_classes:
             # most Perceiver models don't have a typical head like is the case with BERT
-            if model_class in [
-                PerceiverForOpticalFlow,
-                PerceiverForMultimodalAutoencoding,
-                *get_values(MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING),
-                *get_values(MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING),
+            if model_class.__name__ in [
+                "PerceiverForOpticalFlow",
+                "PerceiverForMultimodalAutoencoding",
+                *MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING_NAMES.values(),
+                "PerceiverForImageClassificationLearned",
+                "PerceiverForImageClassificationFourier",
+                "PerceiverForImageClassificationConvProcessing",
+                *MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES.values(),
```

Comment:
Using stuff like `MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES.values()` is not reliable, as an element in `MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES.values()` maybe a list itself. For example, we have

```
        (
            "perceiver",
            (
                "PerceiverForImageClassificationLearned",
                "PerceiverForImageClassificationFourier",
                "PerceiverForImageClassificationConvProcessing",
            ),
        ),
```
in `MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES`.


It's probably better to have something like `_get_names_for_values`: similar to `get_values` in `src/transformers/models/auto/auto_factory.py` but only return names without loading the classes.
 
I could do that in a follow up PR, or any better suggestion.

Reply:
yeah it's faster also to not load the classes and just compare strings